### PR TITLE
feat: containerized backend

### PR DESCRIPTION
**Pull request by C0rydoras**
* Dockerized django backend
* Updated CI and Makefile accordingly
* Added gunicorn

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-      POSTGRES_DB: ${DATABASE_NAME}
+    ports:
+      - "5432:5432"
     volumes:
-      - psqldata:/var/lib/postgresql/data
+      - dbdata:/var/lib/postgresql/data
+  backend:
+    build: ./backend
+    depends_on:
+      - db
```
Comment (open-dynaMIX) [inline docker-compose.yml:13]: Needed for running as unprivileged user:

```
  backend:
    build: ./backend
    user: "${UID:?Set UID env variable to your user id}"
    depends_on:
      - db
```

You need to add your UID to the `.env`-file: `echo UID=$UID > .env`

File: .github/workflows/backend.yaml [hunk 2]
```diff
@@ -8,51 +8,24 @@ on:
...
-          cache: "poetry"
-
-      - name: Install dependencies
-        run: poetry install
-      - name: Lint the Code
-        run: |
-          poetry run black --check .
-          poetry run flake8
+      - name: Lint the backend code
+        run: make lint-backend
```
Comment (open-dynaMIX) [inline .github/workflows/backend.yaml:19]: No frontend linting / tests?

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]
Nice work! :rocket:

File: .github/workflows/backend.yaml (hunk 2 above)
Comment (C0rydoras) [reply .github/workflows/backend.yaml:19]: There is a seperate file for that (frontend.yml)

Comment (open-dynaMIX) [reply .github/workflows/backend.yaml:19]: Ah, in this case, nevermind.

File: backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+RUN pip install -U poetry
+
+COPY pyproject.toml poetry.lock $APP_HOME/
+RUN poetry config virtualenvs.create false; poetry install
+
+COPY . $APP_HOME
+
+EXPOSE 8000
+
+CMD /bin/sh -c "wait-for-it.sh $DATABASE_HOST:${DATABASE_PORT:-5432} -- ./manage.py migrate"
```
[the remark this answers is not in the document]
Comment (C0rydoras) [reply backend/Dockerfile]: In that case I will create a second PR for that.
Which tool would you recommend for this?
gunicorn or hurricane?

[the remark this answers is not in the document]
Comment (C0rydoras) [reply backend/Dockerfile]: I added it in this PR  (i forgot to change branches)

**Pull request merged**